### PR TITLE
Allow specify a custom image visibility for OpenStack

### LINKED ISSUE #117: Openstack baremetal images must be published with `public` visibility
**What happened**:

Recent CVE fixes in OpenStack Glance require Openstack baremetal images to be published with `public` visibility.

### PR DESCRIPTION
**Pull request by MrBatschner**
**What this PR does / why we need it**:

Certain image families in OpenStack must be published with `community` visibility while others must be published with `public` visibility. 

This PR enables GLCI to specify the visibility with which images should be published in the publishing config and sets OpenStack baremetal images to `public`.

**Which issue(s) this PR fixes**:
Fixes #117

**Special notes for your reviewer**:

Please test these changes before merging!
Must be cherry-picked to `rel-1443` and `rel-1592`.

**Release note**:

```feature user
Image visibility for images in OpenStack glance can 
```

### CODE REVIEW

File: glci/model.py [hunk 1]
```diff
@@ -813,6 +813,7 @@ class PublishingTargetOpenstack:
     suffix: typing.Optional[str]
     copy_regions: typing.Optional[list[str]]
     cn_regions: typing.Optional[OpenstackChinaRegions]
+    visibility: str
```
Comment (ccwienk) [inline glci/model.py]: might use enum.StrEnum (will be implicitly cast to str upon usage, but documents allowed / possible values)

File: publishing-cfg.yaml [hunk 2]
```diff
@@ -61,6 +61,7 @@
         region_names:
         - ap-cn-1
         buildresult_bucket: 'replica-cn'
+      visibility: 'community'
```
Comment (ccwienk) [inline publishing-cfg.yaml]: as mentioned, consider defining a StrEnum

File: glci/model.py (hunk 1 above)
Comment (MrBatschner) [reply glci/model.py]: Thanks, addressed.

File: glci/openstack_image.py [hunk 3]
```diff
@@ -115,6 +115,7 @@ def upload_and_publish_image(
     image_properties: dict,
     release: glci.model.OnlineReleaseManifest,
     suffix: str = None,
+    visibility: str = 'community'
```
Comment (ccwienk) [inline glci/openstack_image.py]: use OpenStackVisibility enum-value

**Review by ccwienk — APPROVED** [review]

**Review by ccwienk — APPROVED** [review]

**Pull request merged**